File: .net-ainame/Zhiliao.Ainame.Api/Entities/Art.cs
// -----------------------------------------------------------------------------
// 文章实体：对应表 art；缩略图以 byte[] 映射 MySQL LONGBLOB。
// -----------------------------------------------------------------------------

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Zhiliao.Ainame.Api.Entities;

/// <summary>
/// 后台文章/作品。
/// <para>与 AdminArtController  multipart 上传、Excel 导入共用此表结构。</para>
/// </summary>
[Table("art")]
public class Art
{
    public int Id { get; set; }

    [MaxLength(100)]
    public string Username { get; set; } = "";

    [MaxLength(10)]
    public string Sex { get; set; } = "";

    /// <summary>正文，最长 5000 字符（与校验一致）。</summary>
    [MaxLength(5000)]
    public string Artcontent { get; set; } = "";

    /// <summary>缩略图二进制；列表接口通常不返回以减小流量。</summary>
    public byte[] Thumbnail { get; set; } = [];

    public DateTime CreatedTime { get; set; }
}

File: ainame-j/src/main/java/com/zhiliao/ainame/entity/Art.java
package com.zhiliao.ainame.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Lob;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "art")
public class Art {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "username", length = 100, nullable = false)
    private String username = "";

    @Column(name = "sex", length = 10, nullable = false)
    private String sex = "";

    @Column(name = "artcontent", length = 5000, nullable = false)
    private String artcontent = "";

    @Lob
    @Column(name = "thumbnail", columnDefinition = "LONGBLOB", nullable = false)
    private byte[] thumbnail = new byte[0];

    @Column(name = "created_time", nullable = false)
    private LocalDateTime createdTime;

    @PrePersist
    void prePersist() {
        if (createdTime == null) {
            createdTime = LocalDateTime.now();
        }
    }
}
